DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/Digit.h

```diff
@@ -82,6 +82,40 @@ class Digit
 
   ClassDefNV(Digit, 2);
 };
+
+class DigitHW : public Digit
+{
+ public:
+  DigitHW(UShort_t caliCounter = 0, UShort_t nInjection = 0, UShort_t chargeInjected = 0, UShort_t half = 0, UShort_t disk = 0, UShort_t plane = 0, UShort_t zone = 0, UShort_t cableHW = 0, UShort_t chipindex = 0, UShort_t row = 0, UShort_t col = 0, Int_t charge = 0);
+
+  ~DigitHW() = default;
+
+  /// Get hardware and calibartion info
+  UShort_t getCableHW() const { return mCableHW; }
+  UShort_t getCounter() const { return mCounter; }
+  UShort_t getNInj() const { return mNInj; }
+  UShort_t getChargeInj() const { return mChargeInj; }
+  UShort_t getHalf() const { return mHalf; }
+  UShort_t getDisk() const { return mDisk; }
+  UShort_t getPlane() const { return mPlane; }
+  UShort_t getZone() const { return mZone; }
+
+  /// Set cable HW
+  void setCableHW(UShort_t chw) { mCableHW = chw; }
+
+ private:
+  UShort_t mCounter = 0;
+  UShort_t mNInj = 0;
+  UShort_t mChargeInj = 0;
+  UShort_t mHalf = 0;
+  UShort_t mDisk = 0;
+  UShort_t mPlane = 0;
+  UShort_t mZone = 0;
+  UShort_t mCableHW = 0; ///< Cable HW
```

Comment:
Could you clarify why this information is needed here? I can understand NInj and ChargeInj, although I think the proper place for this info would be the ROFRecord, since it is a property of the whole trigger not the digit.

But such things like Halfdisk, disk, mCableHW etc. are 1) can be derived from the Digit::mChipIndex
2) break the symmetry between the ITS and MFT which we applied quite some efforts to create.

I would consider moving non-redundant info to ROFRecord (I guess we don't need then another ROFRecordHW version) and suppressing all info which can be derived from mChipIndex

Reply:
Hi @shahor02, 
Thanks for your comments. About 2) I believe that the more general class that gives the same interface to ITS and MFT could be subsequently specialised for the behaviour and functionalities that we specifically need for MFT, since the two detectors are not identical. I guess the power of polymorphism lies exactly there. The main point, of course, is that nothing added to get a specialised behaviour that we want to be applied only to MFT alters ITS. Maybe the way we introduce this specialisation is not the best, and we could use some advice there. But the very idea of specialisation should not be thrown away.
About 1): the Digit::mChipIndex is not useable as such as it is a software index that does not allow to easily spot where is the sensor being tested or calibrated in the structure of the detector. During the tests of the MFT for the on-going commissioning phase, we need to have routinely access to the hardware localisation of the sensor, and that should be easily provided by the digit tree instead of having to do this again by a derivation of Digit::mChipIndex in another process following the production of the digit ROOT file. This is a huge gain in term of efficiency during the tests of the detector at Point 2, and therefore is a huge gain of time/reliability for us. I believe that the validation of the detector is a goal important enough to reduce this gap between what is currently available in the software and what we need on a daily basis. We have been using routinely these modifications of the code by @mcoquet642 since the end of last year (and he worked to perfect them before this PR). The behaviour related to these modifications is only activated when the option "--writeHW" is used. So it is transparent for the other users, may they be ITS users or MFT users not directly interested into the digits but to higher level objects built from the digits (typically clusters).

Reply:
Detector specialization is provided be the template parameter classes  https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingITS.h and https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingMFT.h, particularly, you have direct static lookup table from mChipIndex to your specific geometry details https://github.com/AliceO2Group/AliceO2/blob/582b5e47a4eec75872988221cb58da9bf3a85610/Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/ChipMappingMFT.h#L172, there is no particular gain from writing this redundant info.
I think other types of specialization which is not compatible between 2 detectors, should be done in detector specific derived class (after making sure that the partner detector indeed does not need this functionality).